src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/UseCrossPlatformIntrinsics.Fixer.cs

```diff
@@ -0,0 +1,140 @@
+                        UseCrossPlatformIntrinsicsTitle,
+                        c => ReplaceWithCrossPlatformIntrinsicAsync(context.Document, ruleKind, invocation, c),
+                        equivalenceKey: nameof(UseCrossPlatformIntrinsicsFixer)
+                    ),
+                    diagnostic
+                );
+            }
+        }
+
+        private async Task<Document> ReplaceWithCrossPlatformIntrinsicAsync(Document document, RuleKind ruleKind, IInvocationOperation invocation, CancellationToken cancellationToken)
+        {
+            DocumentEditor editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+            SyntaxGenerator generator = editor.Generator;
+
+            SyntaxNode? replacementNode = ruleKind switch
+            {
+                RuleKind.opAddition => ReplaceBinaryOp(invocation, isCommutative: true, generator.AddExpression),
+                RuleKind.opBitwiseAnd => ReplaceBinaryOp(invocation, isCommutative: true, generator.BitwiseAndExpression),
+                RuleKind.opBitwiseOr => ReplaceBinaryOp(invocation, isCommutative: true, generator.BitwiseOrExpression),
+                RuleKind.opDivision => ReplaceBinaryOp(invocation, isCommutative: false, generator.DivideExpression),
+                RuleKind.opExclusiveOr => ReplaceBinaryOp(invocation, isCommutative: true, CreateExclusiveOrExpression),
+                RuleKind.opLeftShift => ReplaceBinaryOp(invocation, isCommutative: false, CreateLeftShiftExpression),
+                RuleKind.opMultiply => ReplaceBinaryOp(invocation, isCommutative: true, generator.MultiplyExpression),
+                RuleKind.opOnesComplement => ReplaceUnaryOp(invocation, generator.BitwiseNotExpression),
+                RuleKind.opRightShift => ReplaceBinaryOp(invocation, isCommutative: false, CreateRightShiftExpression),
+                RuleKind.opSubtraction => ReplaceBinaryOp(invocation, isCommutative: false, generator.SubtractExpression),
+                RuleKind.opUnaryNegation => ReplaceUnaryOp(invocation, generator.NegateExpression),
+                RuleKind.opUnsignedRightShift => ReplaceBinaryOp(invocation, isCommutative: false, CreateUnsignedRightShiftExpression),
+                _ => null,
+            };
+
+            if (replacementNode is not null)
+            {
+                editor.ReplaceNode(invocation.Syntax, replacementNode.WithTriviaFrom(invocation.Syntax));
+                document = document.WithSyntaxRoot(editor.GetChangedRoot());
+            }
+
+            return document;
+
+            static SyntaxNode? ReplaceUnaryOp(IInvocationOperation invocation, Func<SyntaxNode, SyntaxNode?> unaryOpFunc)
+            {
+                if (invocation.Arguments.Length != 1)
+                {
+                    return null;
+                }
+
+                return unaryOpFunc(
+                    invocation.Arguments[0].Value.Syntax
+                );
```

Comment:
```suggestion
                return unaryOpFunc(invocation.Arguments[0].Value.Syntax);
```

It's a nit, but the above is really not the syntax we use :)